src/Cryptol/TypeCheck/Solver/SMT.hs

```diff
@@ -80,13 +84,24 @@ setupSolver s cfg = do
 -- | Start a fresh solver instance
 startSolver :: IO () -> SolverConfig -> IO Solver
 startSolver onExit sCfg =
-   do logger <- if (solverVerbose sCfg) > 0 then SMT.newLogger 0
-
-                                     else return quietLogger
+   do (logger, mbLoggerHdl) <-
+        if (solverVerbose sCfg) > 0
+        then do loggerHdl <-
+                  case solverSmtFile sCfg of
+                    Nothing   -> pure stdout
+                    Just file -> openFile file WriteMode
```

Comment:
I think we need a way to close this file.  Maybe here we can return a `Maybe (IO ())` which will be `Just (hClose hande)` and this should be added to `onExit`?

Reply:
Oh dear, I completely overlooked `onExit` here. I'll fold the `Handle`-closing logic into that per your suggestion.